.kokoro

```diff
@@ -197,8 +197,22 @@ run_bazel_affected() {
       pushd github/repo >> /dev/null
     fi
 
-    TARGET="$(scripts/affected_targets $rule_kind)"
-    
+    # `target_branch` is either the Pull Request destination branch (where it will be merged) or if
+    # none can be determined, defaults to `develop`.
+    if [ -n "$KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH" ]; then
+      target_branch="$KOKORO_GITHUB_PULL_REQUEST_TARGET_BRANCH"
+    else
+      target_branch=develop
+    fi
+
+    # `base_sha` is the merge base of `target_branch` and the current HEAD.
+    base_sha=$(git merge-base "$target_branch" HEAD)
+
+    # `TARGET` is then all bazel targets affected by changes in the commit range between `base_sha`
+    # and `HEAD`.  By default, that becomes all changes between the `develop` branch and the latest
```

Comment:
I could be wrong, but I believe the way this is written, that it's "all changes between the branch point of the target branch and `HEAD`.  

The distinction is a branch that has a very old commit as the merge-base but merged the target branch in on several occasions. The 'affected' files would be all of those changes, including anything that came through on the merges.  

Did I understand the code correctly?

Reply:
> "all changes between the branch point of the target branch and `HEAD`

This is my understanding of what this is doing as well.

> The distinction is a branch that has a very old commit as the merge-base but merged the target branch in on several occasions. The 'affected' files would be all of those changes, including anything that came through on the merges.

I tested this locally to see what would happen:

```bash
> git checkout c06e8198356b641667be4c233304617e52683228
> git checkout -b testbranch
> echo "Modification" >> WORKSPACE
> git commit -a --m "Fake change"

> git merge-base "develop" HEAD
c06e8198356b641667be4c233304617e52683228
> git log --name-only --pretty=oneline --full-index $(git merge-base "develop" HEAD)...HEAD
7d11d7dc5ba1da48595528037d3eacbd61c52f17 (HEAD -> testbranch) Fake change
WORKSPACE

> git merge --no-ff beb7b015ac3b61c758fef24dd98f6a5a3878332e
> git merge-base "develop" HEAD
beb7b015ac3b61c758fef24dd98f6a5a3878332e
> git log --name-only --pretty=oneline --full-index $(git merge-base "develop" HEAD)...HEAD
ec93ede9d346cc78503ee0b5d41e2479edfbcd6c (HEAD -> testbranch) Merge commit 'beb7b015ac3b61c758fef24dd98f6a5a3878332e' into testbranch
7d11d7dc5ba1da48595528037d3eacbd61c52f17 Fake change
WORKSPACE

> git merge --no-ff develop
> git merge-base "develop" HEAD
a635f99d2893ae839cd3d743d35ab98184d6a2f5
> git log --name-only --pretty=oneline --full-index $(git merge-base "develop" HEAD)...HEAD
9b57f663ede6d8e705297051de8c350b5f6396b5 (HEAD -> testbranch) Merge branch 'develop' into testbranch
ec93ede9d346cc78503ee0b5d41e2479edfbcd6c Merge commit 'beb7b015ac3b61c758fef24dd98f6a5a3878332e' into testbranch
7d11d7dc5ba1da48595528037d3eacbd61c52f17 Fake change
WORKSPACE
```

It looks like the command being used in the script correctly includes only files that were modified as part of this branch and ignores changes to files that came in through merge operations.

Reply:
SGTM!